src/wkt/src/duration.rs

```diff
@@ -187,14 +187,14 @@ impl std::convert::From<&Duration> for String {
         if duration.nanos == 0 {
             return format!("{sign}{}s", duration.seconds.abs());
         }
+        let ns = format!("{:09}", duration.nanos.abs());
         if duration.seconds == 0 {
-            let ns = format!("{:09}", duration.nanos.abs());
             return format!("{sign}0.{}s", ns.trim_end_matches('0'));
         }
```

Comment:
looks like we can dispense with this `if`

Reply:
Done.